realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java

```diff
@@ -247,6 +251,86 @@ public void createObjectFromJson_dateAsString() throws JSONException {
         assertEquals(new Date(1000), obj.getColumnDate());
     }
 
+    @Test
+    public void createObjectFromJson_decimal128() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put("columnDecimal128", new Decimal128(BigDecimal.TEN));
+
+        realm.beginTransaction();
+        realm.createObjectFromJson(AllTypes.class, json);
+        realm.commitTransaction();
+
+        AllTypes obj = realm.where(AllTypes.class).findFirst();
+        assertEquals(new Decimal128(BigDecimal.TEN), obj.getColumnDecimal128());
+    }
+
+    @Test
+    public void createUsingJsonStream_decimal128() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put("columnDecimal128", new Decimal128(BigDecimal.TEN));
+
+        realm.beginTransaction();
+        realm.createObjectFromJson(AllTypes.class, json);
+        realm.commitTransaction();
+
+        AllTypes obj = realm.where(AllTypes.class).findFirst();
+        assertEquals(new Decimal128(BigDecimal.TEN), obj.getColumnDecimal128());
+    }
+
+    @Test
+    public void createObjectFromJson_decimal128AsLong() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put("columnDecimal128", -32361122672259149L);
+
+        realm.beginTransaction();
+        realm.createObjectFromJson(AllTypes.class, json);
+        realm.commitTransaction();
+
+        AllTypes obj = realm.where(AllTypes.class).findFirst();
+        assertEquals(new Decimal128(-32361122672259149L), obj.getColumnDecimal128());
+    }
+
+    @Test
+    public void createObjectFromJson_decimal128AsString() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put("columnDecimal128", "32361122672259149");
+
+        realm.beginTransaction();
+        realm.createObjectFromJson(AllTypes.class, json);
+        realm.commitTransaction();
+
+        AllTypes obj = realm.where(AllTypes.class).findFirst();
+        assertEquals(Decimal128.parse("32361122672259149"), obj.getColumnDecimal128());
+    }
+
+    @Test
+    public void createObjectFromJson_objectId() throws JSONException {
+        JSONObject json = new JSONObject();
+        String idHex = TestHelper.generateObjectIdHexString(7);
+        json.put("columnObjectId", new ObjectId(idHex));
```

Comment:
see previous response
>  you could write your own parser which produces a JSONObject/array it could be useful to add directly the Bson types if you know that the underlying consumer system can consume (parse them)